Assets/Scripts/Puzzle/Gate.cs

```diff
@@ -27,18 +27,22 @@ public class Gate : SignalReceiver {
 	}
 
 	public void Toggle() {
-		if (panCamera && camera != null)
+		if (panCamera && camera != null) {
+			if (showRoutine != null)
+				StopCoroutine(showRoutine);
 			StartCoroutine(ShowEvent());
```

Comment:
Stopping this coroutine and restarting it would mess up all the timings because the camera would be somewhere inbetween the player and the gate. I was thinking that a boolean like `isShowing` would do. And just do nothing if it's already running.